LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/Regions/Region.cs

```diff
@@ -152,15 +152,16 @@ public string GetDownstreamDataRate(Rxpk upstreamChannel, int rx1DrOffset = 0)
 
             if (IsValidUpstreamRxpk(upstreamChannel))
             {
-                // If the rx1 offset is a valid value we use it, otherwise we keep answering on normal datar
+                // If the rx1 offset is a valid value we use it, otherwise we report an error
                 if (rx1DrOffset <= RX1DROffsetTable[0].Count - 1)
                 {
                     return DRtoConfiguration[(ushort)RX1DROffsetTable[GetDRFromFreqAndChan(upstreamChannel.Datr)][rx1DrOffset]].configuration;
                 }
                 else
                 {
-                    Logger.Log($"rx1 Dr Offset was not set to a valid value {rx1DrOffset}, defaulting to {upstreamChannel.Datr} datarate", LogLevel.Error);
-                    return upstreamChannel.Datr;
+                    Logger.Log($"RX1 data rate offset was set to an invalid value {rx1DrOffset}; " +
+                        $"maximum allowed offset is {RX1DROffsetTable[0].Count - 1}", LogLevel.Error);
+                    return null;
```

Comment:
I think we could throw a LoRa ProcessingException https://github.com/Azure/iotedge-lorawan-starterkit/pull/713 @bastbu

Reply:
Sounds good. What do you think about the case where `IsValidUpstreamRxpk(upstreamChannel)` is false (where we currently return null)?

Reply:
I think this should be changed to void, if IsValidUpstreamRxpk(upstreamChannel) is false we should also throw

Reply:
I agree, changed the method to return the correct data rate or throw `LoRaProcessingException`.